### PR TITLE
 create end point login #15

### PR DESCRIPTION
**Pull request by mohammedsalah7**

### CODE REVIEW

**Review by ElhamFadel — DISMISSED** [review]

File: server/controllers/users/login.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+const login = async (req, res, next) => {
+  try {
+    const { email, password } = req.body;
+    await loginSchema.validateAsync(req.body);
+
+    const { rows } = await checkEmailQuery(email);
+
+    if (!rows.length) {
+      throw new Error({ message: 'Invalid email or password' });
```
Comment (Mu7ammadAbed) [inline server/controllers/users/login.js:15]:
```suggestion
      return res.status(400).json({ message: 'Invalid email or password' });
```

File: server/controllers/users/login.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    }
+    const compared = await bcrypt.compare(password, rows[0].password);
+    if (!compared) {
+      throw new Error({ message: 'Invalid email or password' });
```
Comment (Mu7ammadAbed) [inline server/controllers/users/login.js:19]:
```suggestion
      return res.status(400).json({ message: 'Invalid email or password' });
```

File: server/controllers/users/login.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  try {
+    const { email, password } = req.body;
+    await loginSchema.validateAsync(req.body);
+
+    const { rows } = await checkEmailQuery(email);
+
+    if (!rows.length) {
+      throw new Error({ message: 'Invalid email or password' });
+    }
+    const compared = await bcrypt.compare(password, rows[0].password);
```
Comment (Mu7ammadAbed) [inline server/controllers/users/login.js:17]: You can make your controller a little bit organizer by splitting these
```suggestion
    }
    
    const compared = await bcrypt.compare(password, rows[0].password);
```

File: server/controllers/users/login.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    if (!compared) {
+      throw new Error({ message: 'Invalid email or password' });
+    }
+    const token = await signToken(email, rows[0].id, rows[0].name, rows[0].phone, rows[0].avater);
+    return res.cookie('token', token).json({ message: 'You are Logged Successfully' });
```
Comment (Mu7ammadAbed) [inline server/controllers/users/login.js:22]: Why would you send all these data in the token ?
you only need the id and the role if there's any
```suggestion
    const token = await signToken(email, rows[0].id);
    
    return res.cookie('token', token).json({ message: 'You are Logged Successfully' });
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]
I think you need to update your branch to the latest development version 
`git pull origin development`
and there's an env requiring in the `connection` file you have to keep it there

**Comment by Mu7ammadAbed** [comment]
You need to update the branch with the development and resolve your conflicts, be aware that there are some changes with the routes file.

**Review by Mu7ammadAbed — DISMISSED** [review]

**Review by ElhamFadel — APPROVED** [review]

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by muhammadharoun — APPROVED** [review]

**Review by ElhamFadel — APPROVED** [review]

**Pull request merged**